src/StudentManagement.java

```diff
@@ -72,22 +116,60 @@ public void searchStudent() throws FileNotFoundException {
         }
     }
     public void readAll() {
+        for (Student student : list) {
+            System.out.println(student);
+        }
+    }
+
+    public void addMembership() {
+        Scanner input = new Scanner(System.in);
+        Student s;
```

Comment:
you should call the variables with proper names, rather than just a single letter